File: server/php/src/Application/Domain/Payments/PaymentDetails/MobilePaymentData.php
<?php

namespace OnlinePayments\ExampleApp\Application\Domain\Payments\PaymentDetails;

final class MobilePaymentData
{
    public ?string $dpan = null;
    public ?string $expiryDate = null;

    public function toArray(): array
    {
        return [
            'dpan' => $this->dpan,
            'expiryDate' => $this->expiryDate,
        ];
    }
}

File: server/ruby/app/business/domain/payments/payment_details/mobile_payment_data.rb
module Business
  module Domain
    module Payments
      module PaymentDetails
        class MobilePaymentData
          attr_accessor :dpan, :expiry_date

          def initialize(dpan: nil, expiry_date: nil)
            @dpan = dpan
            @expiry_date = expiry_date
          end

          def to_h
            {
              'dpan' => @dpan,
              'expiryDate' => @expiry_date
            }
          end
        end
      end
    end
  end
end